src/fsstorage.cc

```diff
@@ -12,6 +12,11 @@
 #include "utils.h"
 
 FSStorage::FSStorage(const StorageConfig& config) : INvStorage(config) {
+  struct stat st;
+  stat(config_.path.c_str(), &st);
+  if ((st.st_mode & (S_IWOTH | S_IWGRP)) != 0) {
```

Comment:
Since we'd have to update from old versions with potentially unsafe permissions, don't we need to try to chmod first and only throw if it fails? Otherwise, it would brick updates on the device.

Reply:
Yes, but task was not to change from unsafe to safe permissions. If I do chmod before every run, what is the reason to check the permissions?

Reply:
If we `chmod` on boot, then we have the issue where someone does nasty things and then we lock them out *after* their changes. There is no way to ensure that current devices don't get locked out. This would have to be in the realase notes.

Reply:
Also is this a config dir? If so, i needs to be `700`. If it's a file, `600`. I can't remember what the bit mask constants are.

Reply:
Yo come to my desk when you see this. I think this will be faster to discuss IRL.